packages/application-extension/src/index.tsx

```diff
@@ -151,6 +151,30 @@ const main: JupyterLabPlugin<void> = {
         return (event as any).returnValue = message;
       }
     });
+
+    let isBusy = false;
+    const favicon = document.querySelector('link[rel="shortcut icon"]') as HTMLLinkElement;
+    const icons = ['favicon-busy-1.ico', 'favicon-busy-3.ico', 'favicon-busy-3.ico'];
+
+    let interval = 0;
+    let i = 0;
+
+    app.serviceManager.sessions.runningChanged.connect((_, sessions) => {
+      const newIsBusy = sessions.map(s => s.kernel.execution_state).indexOf('busy') !== -1;
+      if (newIsBusy !== isBusy) {
+        isBusy = newIsBusy;
+        console.log({isBusy});
+        if (isBusy && !interval) {
+          interval = window.setInterval(() => {
+              favicon.href = `/static/base/images/${icons[i++ % 3]}`;
```

Comment:
Are you animating the favicon here? What does the classic notebook use the other busy indicators for? I only see references to `favicon-busy-1.ico`.

Reply:
Ah, they used to use the animated favicon, but then they removed it: https://github.com/jupyter/notebook/pull/2675/files For some reason I was found the commit adding the animated state and didn't see that they undid it.

I will change it not to animate.